scipy/optimize/tests/test__linprog_clean_inputs.py

```diff
@@ -237,33 +240,72 @@ def test__clean_inputs3():
     assert_raises(ValueError, _clean_inputs, lp._replace(bounds=[(1, 2), (1, 2, 2)]))
     assert_raises(ValueError, _clean_inputs, lp._replace(bounds=[(1, 2), (1, 2), (1, 2)]))
 
+    lp = _LPProblem(c=[1, 2, 3, 4])
+
+    assert_raises(ValueError, _clean_inputs, lp._replace(bounds=[(1, 2, 3, 4), (1, 2, 3, 4)]))
+
 
 def test_good_bounds():
     lp = _LPProblem(c=[1, 2])
 
     lp_cleaned = _clean_inputs(lp)  # lp.bounds is None by default
-    assert_(lp_cleaned.bounds == [(0, None)] * 2, "")
+    assert_(np.all(lp_cleaned.bounds == [(0, np.inf)] * 2), "")
 
     lp_cleaned = _clean_inputs(lp._replace(bounds=(1, 2)))
-    assert_(lp_cleaned.bounds == [(1, 2)] * 2, "")
+    assert_(np.all(lp_cleaned.bounds == [(1, 2)] * 2), "")
 
     lp_cleaned = _clean_inputs(lp._replace(bounds=[(1, 2)]))
-    assert_(lp_cleaned.bounds == [(1, 2)] * 2, "")
+    assert_(np.all(lp_cleaned.bounds == [(1, 2)] * 2), "")
+
+    lp_cleaned = _clean_inputs(lp._replace(bounds=[(1, None)]))
+    assert_(np.all(lp_cleaned.bounds == [(1, np.inf)] * 2), "")
+
+    lp_cleaned = _clean_inputs(lp._replace(bounds=[(None, 1)]))
+    assert_(np.all(lp_cleaned.bounds == [(-np.inf, 1)] * 2), "")
 
-    lp_cleaned = _clean_inputs(lp._replace(bounds=[(1, np.inf)]))
-    assert_(lp_cleaned.bounds == [(1, None)] * 2, "")
+    lp_cleaned = _clean_inputs(lp._replace(bounds=[(None, None), (-np.inf, None)]))
+    assert_(np.all(lp_cleaned.bounds == [(-np.inf, np.inf)] * 2), "")
 
-    lp_cleaned = _clean_inputs(lp._replace(bounds=[(-np.inf, 1)]))
-    assert_(lp_cleaned.bounds == [(None, 1)] * 2, "")
+    lp = _LPProblem(c=[1, 2, 3, 4])
+
+    lp_cleaned = _clean_inputs(lp)  # lp.bounds is None by default
+    assert_(np.all(lp_cleaned.bounds == [(0, np.inf)] * 4), "")
+
+    lp_cleaned = _clean_inputs(lp._replace(bounds=(1, 2)))
+    assert_(np.all(lp_cleaned.bounds == [(1, 2)] * 4), "")
 
-    lp_cleaned = _clean_inputs(lp._replace(bounds=[(-np.inf, np.inf), (-np.inf, np.inf)]))
-    assert_(lp_cleaned.bounds == [(None, None)] * 2, "")
+    lp_cleaned = _clean_inputs(lp._replace(bounds=[(1, 2)]))
+    assert_(np.all(lp_cleaned.bounds == [(1, 2)] * 4), "")
+
+    lp_cleaned = _clean_inputs(lp._replace(bounds=[(1, None)]))
+    assert_(np.all(lp_cleaned.bounds == [(1, np.inf)] * 4), "")
+
+    lp_cleaned = _clean_inputs(lp._replace(bounds=[(None, 1)]))
+    assert_(np.all(lp_cleaned.bounds == [(-np.inf, 1)] * 4), "")
+
+    lp_cleaned = _clean_inputs(lp._replace(bounds=[(None, None), (-np.inf, None), (None, np.inf), (-np.inf, np.inf)]))
+    assert_(np.all(lp_cleaned.bounds == [(-np.inf, np.inf)] * 4), "")
+
+def test_bounds_branches():
+
+    lp = _LPProblem(c=[1, 2, 3, 4])
+
+    # branch 1
+    lp_cleaned = _clean_inputs(lp._replace(bounds=[(1, 2),(1, 2),(1, 2),(1, 2)]))
+    # branch 2
+    lp_cleaned = _clean_inputs(lp._replace(bounds=[(1, 2)]))
+    lp_cleaned = _clean_inputs(lp._replace(bounds=[[1],[2]]))
```

Comment:
What is the purpose of this if nothing is checked?

Reply:
Forgot to remove this test. I tried to set up a test to check whether each of the if branches was reached appropriately, but that is something you cannot check from the outside without adding extra code.